sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1337,6 +1350,35 @@ case class Cast(
     }
   }
 
+  private[this] def castToTimeCode(
+      from: DataType,
+      ctx: CodegenContext): CastFunction = {
+    from match {
+      case _: StringType =>
+        val longOpt = ctx.freshVariable("longOpt", classOf[Option[Long]])
+        (c, evPrim, evNull) =>
+          if (ansiEnabled) {
+            val errorContext = getContextOrNullCode(ctx)
+            code"""
+              $evPrim = $dateTimeUtilsCls.stringToTimeAnsi($c, $errorContext);
+            """
+          } else {
+            code"""
+              scala.Option<Long> $longOpt =
+                org.apache.spark.sql.catalyst.util.DateTimeUtils.stringToTime($c);
```

Comment:
Can also use `$dateTimeUtilsCls.stringToTime($c);` ?

Reply:
Fixed, and changed other places too. Please, review the PR: https://github.com/apache/spark/pull/50251